lib/vfs/src/filesystems.rs

```diff
@@ -0,0 +1,106 @@
+use crate::FileSystem;
+
+/// A chain of one or more [`FileSystem`]s.
+pub trait FileSystems<'a>: 'a {
```

Comment:
Is this primarily just here to avoid writing `IntoIterator` bounds on the `OverlayFs`?

Maybe there is a reason I'm missing, but seems like this could just be cut.

I'd rather not take this into the public API - we'll probably want to be conservative with this specific crate.

Reply:
It also allows you to use tuples instead of forcing you to store a `Vec<Box<Dyn FileSystem>>` when the secondaries are homogeneous, which would introduce two extra levels of indirection. 

The "one read-write and a list of read-only" semantics are also what we decided we wanted when Syrus and I were discussing during our Planning & Alignment meeting.